scipy/stats/_stats_py.py

```diff
@@ -8852,9 +8850,9 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
     rankx_mean = np.mean(rankx)
     ranky_mean = np.mean(ranky)
 
-    Sx = np.sum(np.power(rankcx - rankx - rankcx_mean + rankx_mean, 2.0))
+    Sx = np.linalg.norm(rankcx - rankx - rankcx_mean + rankx_mean)**2
```

Comment:
Note that input to `brunnermunzel` is always 1d.

```suggestion
    temp = rankcx - rankx - rankcx_mean + rankx_mean
    Sx = np_vecdot(temp, temp)
```